### PR TITLE
speaker recognition and verification docs

### PR DESCRIPTION
**Pull request by nithinraok**
Signed-off-by: Nithin Rao Koluguri <nithinrao.koluguri@gmail.com>
* Adds speaker recognition and verification docs

### CODE REVIEW

File: docs/source/asr/speaker_recognition/intro.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+Speaker Recognition (SR)
+========================
+
+Speaker Recognition (SR) is a broad research area which solves two major tasks: speaker identification (who is speaking?) and speaker verification (is the speaker who they claim to be?). 
+We focus on far-field, text-independent speaker recognition when the identity of the speaker is based on how the speech is spoken, not necessarily in what is being said. 
+Typically such SR systems operate on unconstrained speech utterances, which are converted into vectors of fixed length, called speaker embeddings. 
+Speaker embeddings are also used in automatic speech recognition (ASR) and speech synthesis.
+
+As the goal of most speaker related systems is to get good speaker level embeddings that could help distinguish from other speakers, we shall first train these embeddings in end-to-end manner optimizing the QuatzNet based encoder model on cross-entropy loss. 
+We modify the decoder to get these fixed size embeddings irrespective of the length of ithe input audio and employ a mean and variance based statistics pooling method to grab these embeddings.
```
Comment (titu1994) [inline docs/source/asr/speaker_recognition/intro.rst:10]: `length of ithe` <- fix

File: docs/source/asr/speaker_recognition/intro.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+   
+In Speaker Identification we typically train on a larger set with cross-entrophy loss and finetune later on your preferred set of labels where one would want to classify only known set of speakers.
```
Comment (titu1994) [inline docs/source/asr/speaker_recognition/intro.rst:12]: a "larger set" of what? Clear up sentence

File: docs/source/asr/speaker_recognition/intro.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Typically such SR systems operate on unconstrained speech utterances, which are converted into vectors of fixed length, called speaker embeddings. 
+Speaker embeddings are also used in automatic speech recognition (ASR) and speech synthesis.
+
+As the goal of most speaker related systems is to get good speaker level embeddings that could help distinguish from other speakers, we shall first train these embeddings in end-to-end manner optimizing the QuatzNet based encoder model on cross-entropy loss. 
+We modify the decoder to get these fixed size embeddings irrespective of the length of ithe input audio and employ a mean and variance based statistics pooling method to grab these embeddings.
+   
+In Speaker Identification we typically train on a larger set with cross-entrophy loss and finetune later on your preferred set of labels where one would want to classify only known set of speakers. 
+And in Speaker verification we train with Angular Softmax loss and compare embedings extracted from one audio file coming from a single speaker with 
+embeddings extracted from another file of same or another speaker by employing backend scoring techniques like cosine similarity or PLDA.
```
Comment (titu1994) [inline docs/source/asr/speaker_recognition/intro.rst:14]: Add reference to PLDA (at least write the full name first)

File: docs/source/asr/speaker_recognition/models.rst [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+The QuartzNet encoder used for speaker embeddings shown in figure below has the following structure: a QuartzNet BxR
+model has B blocks, each with R sub-blocks. Each sub-block applies the following operations: a 1D convolution, batch norm, ReLU, and dropout. All sub-blocks in a block have the same number of output channels. These blocks are connected with residual connections. We use QuartzNet with 3 blocks, 2 sub-blocks, and 512 channels, as the Encoder for Speaker Embeddings. All conv layers have stride 1 and dilation 1.
+
+
+    .. image:: images/ICASPP_SpeakerNet.png
+        :align: center
+        :alt: speakernet model
+        :scale: 40%
+
+Top level acoustic Features, E obtained from the output of
```
Comment (titu1994) [inline docs/source/asr/speaker_recognition/models.rst:29]: "Top level acoustic Features, E obtained" ?

File: docs/source/asr/speaker_recognition/results.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  import nemo.collections.asr as nemo_asr
+  model = nemo_asr.models.<MODEL_BASE_CLASS>.restore_from(restore_path="<path/to/checkpoint/file.nemo>")
+
+Where the model base class is the ASR model class of the original checkpoint, or the general `ASRModel` class.
+
+NGC Pretrained Checkpoints
+--------------------------
+
+The SpeakerNet-ASR collection has checkpoints of several models trained on various datasets for a variety of tasks.
+These checkpoints are obtainable via NGC at `NeMo Speech collection <https://ngc.nvidia.com/catalog/models/nvidia:nemospeechmodels>`_.
```
Comment (titu1994) [inline docs/source/asr/speaker_recognition/results.rst:34]: Why is it linking to old nemospeechmodels directory? Point to the new model cards

File: docs/source/index.rst [hunk 6]
```diff
@@ -11,12 +11,16 @@ NVIDIA NeMo Developer Guide
    introduction/tutorials
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 8
```
Comment (titu1994) [inline docs/source/index.rst:14]: Why increase this? Revert

**Review by titu1994 — CHANGES_REQUESTED** [review]
Overall looks great. Minor grammar fixes here and there.

File: docs/source/asr/speaker_recognition/results.rst (hunk 5 above)
Comment (nithinraok) [reply docs/source/asr/speaker_recognition/results.rst:34]: Do you know where are all speech models available for public? All we have are nvstaging links.

File: docs/source/asr/speaker_recognition/models.rst (hunk 4 above)
Comment (nithinraok) [reply docs/source/asr/speaker_recognition/models.rst:29]: symbol to represent those features is E.

**Review by titu1994 — APPROVED** [review]
LGTM, but wait @fayejf review before merge

**Review by fayejf — APPROVED** [review]
LGTM.

**Pull request merged**